pkg/controllers/binding/binding_controller.go

```diff
@@ -110,7 +112,7 @@ func (c *ResourceBindingController) syncBinding(binding *workv1alpha2.ResourceBi
 		return controllerruntime.Result{Requeue: true}, err
 	}
 	var errs []error
-	err = ensureWork(c.Client, workload, c.OverrideManager, binding, apiextensionsv1.NamespaceScoped)
+	err = ensureWork(c.Client, c.ResourceInterpreter, workload, c.OverrideManager, binding, apiextensionsv1.NamespaceScoped)
```

Comment:
Too much parameters required by `ensureWork` now, maybe you can help review the #1145.

Reply:
Yes but these are all needed, IMHO #1145 just dived one function into several parts, it doesn't solve the too much parameters problem, I think this function will be clearer and more unified after we remove the deprecated RSP API.

Reply:
OK.